File: Daily Challanges/2026-03-March/2026-03-31.py
def wake_up_alarm(alarm: str, wake: str) -> str:
    ah, am = map(int, alarm.split(":"))
    wh, wm = map(int, wake.split(":"))

    alarm_minutes = ah * 60 + am
    wake_minutes = wh * 60 + wm
    diff = wake_minutes - alarm_minutes

    if diff < 0:
        return "early"
    elif diff <= 10:
        return "on time"
    else:
        return "late"

File: Daily Challanges/2026-03-March/2026-03-31.js
function wakeUpAlarm(alarm, wake) {
  const [ah, am] = alarm.split(":").map(Number);
  const [wh, wm] = wake.split(":").map(Number);

  const alarmMinutes = ah * 60 + am;
  const wakeMinutes = wh * 60 + wm;

  const diff = wakeMinutes - alarmMinutes;

  if (diff < 0) return "early";
  if (diff <= 10) return "on time";
  return "late";
}
